packages/react-aria-components/src/Button.tsx

```diff
@@ -78,31 +104,87 @@ export const ButtonContext = createContext<ContextValue<ButtonContextValue, HTML
 
 function Button(props: ButtonProps, ref: ForwardedRef<HTMLButtonElement>) {
   [props, ref] = useContextProps(props, ref, ButtonContext);
+  props = disablePendingProps(props);
   let ctx = props as ButtonContextValue;
+  let {isPending} = ctx;
   let {buttonProps, isPressed} = useButton(props, ref);
   let {focusProps, isFocused, isFocusVisible} = useFocusRing(props);
   let {hoverProps, isHovered} = useHover(props);
+  let renderValues = {isHovered, isPressed, isFocused, isFocusVisible, isDisabled: props.isDisabled || false, isPending};
```

Comment:
We should make sure hover and press styles don't apply when isPending so it acts like disabled (except focusable).